jobs/export_process/worker.go

```diff
@@ -44,15 +45,21 @@ func MakeWorker(jobServer *jobs.JobServer, app AppIface) model.Worker {
 
 		errCh := make(chan *model.AppError, 1)
 		go func() {
-			defer close(errCh)
+			defer func() {
+				close(errCh)
+
+				// we close the reader here to prevent a deadlock when the bulk exporter tries to
+				// write into the pipe while app.WriteFile has already returned. The error will be
+				// returned by the writer part of the pipe when app.BulkExport tries to call
+				// wr.Write() on it.
+				rd.CloseWithError(errors.New("Worker: writer has been closed"))
```

Comment:
Also possible, we can close it either way. closing it from both sides makes no difference. I wanted to catch `panic`s as well, here. 2/5

Reply:
`panic`s are a bit of a special case I think . We wouldn't even know if `rd` is in a position where it can even be closed or not. I'd prefer to let `panic`s crash loudly.

I understand it makes no difference, but I think closing it always, makes it a general case rather than a special case of only when `app.WriteFile` returns early. So readability wise, it makes things much clear IMO.

WDYT?

Reply:
Yes, that's reasonable. I'll change it.